components/NavigationBar/src/MDCNavigationBar.m

```diff
@@ -460,10 +460,15 @@ - (CGRect)mdc_frameAlignedVertically:(CGRect)frame
 - (CGRect)mdc_frameAlignedHorizontally:(CGRect)frame
                              alignment:(MDCNavigationBarTitleAlignment)alignment {
   switch (alignment) {
-    case MDCNavigationBarTitleAlignmentCenter:
-      return CGRectMake(CGRectGetMaxX(self.bounds) / 2 - frame.size.width / 2, frame.origin.y,
-                        frame.size.width, frame.size.height);
-
+      // Center align title if desired unless leading icons will overlap frame
+    case MDCNavigationBarTitleAlignmentCenter: {
+      CGFloat xOrigin = CGRectGetMaxX(self.bounds) / 2 - frame.size.width / 2;
```

Comment:
Avoid frame.size or frame.origin unless you are positive the frame has been standardized.  (And in that case, document that it is).
s/frame.size.width/CGRectGetWidth(frame)/
(for height as well)